src/pybind/mgr/cephadm/tests/test_upgrade.py

```diff
@@ -164,3 +164,48 @@ def test_enough_mds_for_ok_to_stop(get, get_daemons_by_service, cephadm_module:
     get_daemons_by_service.side_effect = [[DaemonDescription(), DaemonDescription()]]
     assert cephadm_module.upgrade._enough_mds_for_ok_to_stop(
         DaemonDescription(daemon_type='mds', daemon_id='myfs.test.host1.gfknd', service_name='mds.myfs.test'))
+
+
+@pytest.mark.parametrize("tags, current_version, result",
+                         [
+                             # several candidate versions (from different major versions)
+                             (
+                                 [
+                                     'v17.1.0',
+                                     'v16.2.7',
+                                     'v16.2.6',
+                                     'v16.2.5',
+                                     'v16.1.4',
+                                     'v16.1.3',
+                                     'v15.2.0',
+                                 ],
+                                 (16, 1, '16.1.0'),
+                                 ['17.1.0', '16.2.7', '16.2.6', '16.2.5']
```

Comment:
I think we'd want to see 16.1.3 and 16.1.4 here. This implies somebody on 16.2.0 (initial pacific release) would not see 16.2.7 (most recent stable pacific release) which is not good. Just generally same major and minor version but higher z stream should definitely be present. Honestly, even if it's a lower z stream, if the major and minor version match we should probably show it.

Reply:
Updated the code to also show all the releases with the same major/minor version (right now we don't check z stream... so we will show minor z streams if available).